core/devmode/src/main/java/io/quarkus/dev/CompilationProvider.java

```diff
@@ -24,19 +26,22 @@
         private final File projectDirectory;
         private final File sourceDirectory;
         private final File outputDirectory;
+        private final Charset sourceEncoding;
 
         public Context(
                 String name,
                 Set<File> classpath,
                 File projectDirectory,
                 File sourceDirectory,
-                File outputDirectory) {
+                File outputDirectory,
+                String sourceEncoding) {
 
             this.name = name;
             this.classpath = classpath;
             this.projectDirectory = projectDirectory;
             this.sourceDirectory = sourceDirectory;
             this.outputDirectory = outputDirectory;
+            this.sourceEncoding = sourceEncoding == null ? StandardCharsets.UTF_8 : Charset.forName(sourceEncoding);
```

Comment:
`sourceEncoding ` can't be typed sooner to `Charset` because `DevModeContext` is serialized in the Maven and Gradle plugins.